server/src/main/java/org/opensearch/action/admin/cluster/remotestore/stats/RemoteStoreStatsResponse.java

```diff
@@ -41,30 +50,85 @@ public RemoteStoreStatsResponse(
         List<DefaultShardOperationFailedException> shardFailures
     ) {
         super(totalShards, successfulShards, failedShards, shardFailures);
-        this.shards = shards;
+        this.remoteStoreStats = shards;
     }
 
-    public RemoteStoreStats[] getShards() {
-        return this.shards;
+    public RemoteStoreStats[] getRemoteStoreStats() {
+        return this.remoteStoreStats;
     }
 
     public RemoteStoreStats getAt(int position) {
-        return shards[position];
+        return remoteStoreStats[position];
+    }
+
+    public Map<String, Map<Integer, List<RemoteStoreStats>>> groupByIndexAndShards() {
+        if (indexWiseStats != null) {
+            return indexWiseStats;
+        }
+        Set<String> indexNames = new HashSet<>();
+        for (RemoteStoreStats shardStat : remoteStoreStats) {
+            String indexName = shardStat.getShardRouting().getIndexName();
+            indexNames.add(indexName);
+        }
+
+        Map<String, Map<Integer, List<RemoteStoreStats>>> indexWiseStats = new HashMap<>();
+        for (String indexName : indexNames) {
+            Set<RemoteStoreStats> perIndexStats = new HashSet<>();
+            for (RemoteStoreStats shardStat : remoteStoreStats) {
+                if (shardStat.getShardRouting().getIndexName().equals(indexName)) {
+                    perIndexStats.add(shardStat);
+                }
+            }
+            indexWiseStats.put(indexName, groupByShards(perIndexStats));
+        }
+        this.indexWiseStats = indexWiseStats;
+        return indexWiseStats;
+    }
+
+    private Map<Integer, List<RemoteStoreStats>> groupByShards(Set<RemoteStoreStats> perIndexStats) {
+        Map<Integer, List<RemoteStoreStats>> shardStats = new HashMap<>();
+        Set<Integer> shardIds = new HashSet<>();
+        for (RemoteStoreStats eachShardStats : perIndexStats) {
+            int shardId = eachShardStats.getShardRouting().getId();
+            shardIds.add(shardId);
+        }
+
+        for (Integer shardId : shardIds) {
+            List<RemoteStoreStats> stats = new ArrayList<>();
+            for (RemoteStoreStats perShardStats : perIndexStats) {
+                if (perShardStats.getShardRouting().getId() == shardId) {
+                    stats.add(perShardStats);
+                }
+            }
+            shardStats.put(shardId, stats);
+        }
+        return shardStats;
     }
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         super.writeTo(out);
-        out.writeArray(shards);
+        out.writeArray(remoteStoreStats);
     }
 
     @Override
     protected void addCustomXContentFields(XContentBuilder builder, Params params) throws IOException {
-        builder.startArray("stats");
-        for (RemoteStoreStats shard : shards) {
-            shard.toXContent(builder, params);
+        groupByIndexAndShards();
+        builder.startObject("indices");
```

Comment:
nit - can we use any existing string constant or create a new one? Same comment for "shards" as well.

Reply:
Will check on this.